drivers/bridge/bridge.go

```diff
@@ -371,44 +372,52 @@ func (d *driver) DeleteNetwork(nid types.UUID) error {
 	return err
 }
 
-func (d *driver) CreateEndpoint(nid, eid types.UUID, epOptions map[string]interface{}) (*sandbox.Info, error) {
+func (d *driver) CreateEndpoint(nid, eid types.UUID, epInfo driverapi.EndpointInfo, epOptions map[string]interface{}) error {
 	var (
 		ipv6Addr *net.IPNet
 		err      error
 	)
 
+	if epInfo == nil {
+		return errors.New("invalid endpoint info passed")
```

Comment:
@aboch But with your upcoming changes I don't think we need to define a new error type for every error that we return. We need to do that only for errors which we want the caller to act in a special way. In fact we can have a libnetwork specific `New` and `Errorf` methods which gives us that.